### ISSUE TITLE
tqdm + logging, message containing line feed

### ISSUE THREAD

**Issue by Erawpalassalg** [tqdm/tqdm#724]
tqdm v. 4.0.0
Python v. 3.7.0

When using tqdm with logging, configured as @casperdcl suggests it in #313 **BUT** with a line feed in the message, tqdm eats the new line.

Test code:
```
import tqdm
import logging
import time


class TqdmStream(object):
  @classmethod
  def write(_, msg):
    tqdm.tqdm.write(msg, end='')

  # is this required?
  # @classmethod
  # def flush(_):
  #   pass


if __name__ == "__main__":
  logging.basicConfig(level=logging.DEBUG, stream=TqdmStream)

  log = logging.getLogger(__name__)
  log.info("loop")
  # for x in tqdm.trange(1, 101, mininterval=10):
  for x in tqdm.trange(1, 101):
    time.sleep(.2)
    if not x % 20:
      log.debug("in loop %d\nnew line" % x)
  log.info("done")
```

Now, several test cases:
- If a `\n` is added to the end of the message, the text now gets displayed properly (with the line feed at the end, but we asked for it)
- If we use `end="\n"` instead of `end=""`, the message gets displayed properly but with **two** line feeds at the end. 
- passing `end="test"` shows that tqdm seems to have an issue with stdout buffering, since it appends "test" to the next logging line (probably an issue on itself, not sure how it's related to this issue)

One might think that this is a flushing issue, but adding the following solves the 3rd test case (while breaking the progress bar though), while doing nothing for the 2nd one (not even breaking the progress bar).
```
import sys

class TqdmStream(object):
  @classmethod
  def write(_, msg):
    tqdm.tqdm.write(msg, end='')

  @classmethod
  def flush(_):
    sys.sdout.flush()
```

**Comment by evandrocoan** [comment]
> **BUT** with a line feed in the message, tqdm eats the new line.

I did not understand this. I created simplified version of your example, reducing the time and loop range for fast results:
```python
import tqdm
import sys
import logging
import time

class DummyTqdmFile(object):
    """ Dummy file-like that will write to tqdm
    https://github.com/tqdm/tqdm/issues/313
    """
    file = None

    def __init__(self, file):
        self.file = file

    def write(self, x):
        # Avoid print() second call (useless \n)
        if len(x.rstrip()) > 0:
            tqdm.tqdm.write(x, file=self.file, end='')

    def flush(self):
        return getattr(self.file, "flush", lambda: None)()

if __name__ == "__main__":
    logging.basicConfig(level=logging.DEBUG, stream=DummyTqdmFile(sys.stderr))
    log = logging.getLogger(__name__)
    log.info("loop")

    for x in tqdm.trange(1, 16):
        time.sleep(.2)

        if not x % 5:
            log.debug("in loop %d\nnew line" % x)

    log.info("done")
```


When I run your example here under Cygwin mintty.exe, with Anaconda Python 3.7.2 and tqdm  4.23.4 I got this:

![zoom](https://user-images.githubusercontent.com/5332158/57200310-dc97f300-6f60-11e9-8d05-e8dbdb905148.gif)

Is what is wrong with this results?

Running the same example inside Sublime Text, I got this:

![zoom](https://user-images.githubusercontent.com/5332158/57200337-1bc64400-6f61-11e9-87f2-8bca64ae1f0c.gif)

Running the same example inside Windows cmd.exe, I got this:

![zoom2](https://user-images.githubusercontent.com/5332158/57200419-35b45680-6f62-11e9-8591-c441b9fa2fdd.gif)

**Comment by casperdcl** [comment]
yes well `tqdm.write` shpukd be renamed `tqdm.print`. it tries to be perhaps a little too intelligent.

**Comment by Erawpalassalg** [comment]
After a few more tests the issue seems to be xterm related and does not reproduce with any term emulator.

**Comment by alex-tsukanov** [comment]
Experiencing similar problems inside PyCharm on Ubuntu 18.04. Would love to see a fix :)

**Comment by golyalpha** [comment]
I have a slightly different issue, with tqdm progressbars getting reordered in an unusual way.

Here's an asciinema recording of the behaviour: https://asciinema.org/a/w2DZIs0uMov5VERFJrlk8nw6j (sorry for the relatively long downloads)
